torch/utils/data/distributed.py

```diff
@@ -47,7 +47,7 @@ class DistributedSampler(Sampler):
         ...     train(loader)
     """
 
-    def __init__(self, dataset, num_replicas=None, rank=None, shuffle=True, seed=0):
+    def __init__(self, dataset, num_replicas=None, rank=None, shuffle=False, seed=0):
```

Comment:
Does the default change present a backwards compatibility concern? This means previously written code like: 
```
sampler = DistributedSampler()
for epoch in epochs:
    sampler.set_epoch(epoch)
```

won't work as expected since shuffle would be false now.

Reply:
This is a very reasonable concern. Maybe we can change the default `seed` to `1`, so the data is truly shuffled on epoch 0.